### ISSUE TITLE
Correct App loading progress. HowTo?

### ISSUE THREAD

**Issue by AlexVK79** [emscripten-core/emscripten#4588]
We are using emsdk-1.35.0 to port our C++ game and engine to HTML5/WebGL.

Our engine packs all game assets into a single .pak file (which is very similar to ZIP file).
We use --preload-file out_game.pak to map this file into MEMFS and then use
our legacy C++ code to work with our pak(ZIP) file.

1). It all works pretty good, however when testing the game over Web
with the default .html template there is a HUGE 
(up to 2-3 minutes on 5Mbps LTE internet for a 5MB game.pak) 
delay between the spinning wheel
saying "All Downloads Complete." and the actual game startup.
That same delay is almost non-existent when we reload the page 2nd, 3rd, etc time.

Comparing to how fast the main game.js (2MB) is beeing downloaded,
it looks like our 5MB pak file is beeing downloaded much-much slower...
Or is there some additional work emscripten does (and caches) during the 1st stratup ?

2). Is there a way (using --pre-js script or some other way) to calculate and show some 
progress bar with percentage numbers, while emscripten downloads the --preload-file
asset(s) ?

Basically we want to be able to have something like a simple  Flash preloader  script
- show a background picture and CORRECT progress bar while the game and all data donwload.

**Comment by juj** [comment]
Try using the browser Network tab to profile that it's actually the network download of the package that is the slow process. On second run it's possible that the browser has cached the package so it doesn't download it anymore.

In Firefox you'll see a web page log message that shows how long it took to compile the asm.js code on the page. This is the other big chunk of executing that occurs at startup.

If neither of those explain the time spent, after that, it should come down to whatever the page itself does in prerun and main() steps. A JS side profiler should show if something is taking up time on the page itself at startup.

If your downloads are part of the --preload-file/file_packager.py mechanism, it provides progress on the `Module` object from here: https://github.com/kripken/emscripten/blob/master/tools/file_packager.py#L670. It also calls `Module.setStatus` to report progress: https://github.com/kripken/emscripten/blob/master/tools/file_packager.py#L687. Perhaps those will be useful?
